pkg/controller/lifecycle/actuator.go

```diff
@@ -153,18 +155,30 @@ func (a *actuator) Reconcile(ctx context.Context, logger logr.Logger, ex *extens
 		image.Tag = ptr.To[string](version.Get().GitVersion)
 	}
 
-	lakomConfig, err := yaml.JSONToYAML(a.serviceConfig.CosignPublicKeys.Raw)
+	gardenerPublicKeys, err := yaml.JSONToYAML(a.serviceConfig.CosignPublicKeys.Raw)
 	if err != nil {
 		return fmt.Errorf("failed to convert lakom config from json to yaml, %w", err)
 	}
 
+	var clientPublicKeys []byte
+	if lakomProviderConfig.TrustedKeysResourceName != nil {
+		var err error
+		clientPublicKeys, err = getClientKeys(ctx, a.client, cluster.Shoot.Spec.Resources, *lakomProviderConfig.TrustedKeysResourceName, namespace)
+		if err != nil {
+			return fmt.Errorf("failed to get the additional keys: %w", err)
+		}
+	}
+
+	lakomPublicKeys := gardenerPublicKeys
+	lakomPublicKeys = append(lakomPublicKeys, clientPublicKeys...)
```

Comment:
```suggestion
	lakomPublicKeys := append(gardenerPublicKeys, clientPublicKeys...)
```

Reply:
This was actually the initial implementation. But one of the go linters complained about assigning the result of append to a different slice. Ref https://concourse.ci.gardener.cloud/builds/357267772

I'm not fond of the way it's now, but this was the easiest/cleanest alternative I could think of

Reply:
I thought about renaming gardenerPublicKeys to lakomPublicKeys, but that conflicts with my wishes here https://github.com/gardener/gardener-extension-shoot-lakom-service/pull/112#discussion_r1810375531 😃
So I'd rather we leave it as it is